### PR TITLE
Allow adding "ankisearch:" links in external programs.

### PR DESCRIPTION
**Pull request by tatsumoto-ren**
This PR lets any external program create "ankisearch:" links. By clicking on such link, the Anki Browser opens with the specified query.

https://user-images.githubusercontent.com/69171671/226745860-e9eef053-518b-471b-8154-126c9e0755c6.mp4

This can be useful to check if you already have a certain word added to Anki.

### CODE REVIEW

File: ankiconnector.cpp [hunk 1]
```diff
@@ -50,6 +50,27 @@ void AnkiConnector::sendToAnki( QString const & word, QString const & text, QStr
                                        Utils::json2String( fields ) );
 
 //  qDebug().noquote() << postData;
+    postToAnki( postData );
+}
+
+void AnkiConnector::ankiSearch( QString const & word )
+{
+    QString postTemplate = QString(
+      "{"
```
Comment (xiaoyifang) [inline ankiconnector.cpp:59]: can
use rawstring

Comment (tatsumoto-ren) [reply ankiconnector.cpp]: good idea

**Comment by xiaoyifang** [comment]
what will happen,if no anki installed and click on the ankisearch link

File: ankiconnector.cpp [hunk 2]
```diff
@@ -50,6 +50,23 @@ void AnkiConnector::sendToAnki( QString const & word, QString const & text, QStr
                                        Utils::json2String( fields ) );
 
 //  qDebug().noquote() << postData;
+    postToAnki( postData );
+}
+
+void AnkiConnector::ankiSearch( QString const & word )
+{
+    QString postTemplate = R"anki({
```
Comment (xiaoyifang) [inline ankiconnector.cpp:58]: if anki connect is not enabled ,can return directly?

Comment (tatsumoto-ren) [reply ankiconnector.cpp:58]: > what will happen,if no anki installed and click on the ankisearch link

If Ankiconnect isn't running or if Anki isn't runnning, it prints "connection refused" at the bottom.

**Comment by tatsumoto-ren** [comment]
> what will happen,if no anki installed and click on the ankisearch link

If Ankiconnect isn't running or if Anki isn't runnning, it prints "connection refused" at the bottom.

File: ankiconnector.cpp (hunk 2 above)
Comment (tatsumoto-ren) [reply ankiconnector.cpp:58]: > if anki connect is not enabled ,can return directly?

Cancel the operation if ankiconnect isn't enabled in settings? i'll look into it.

Comment (tatsumoto-ren) [reply ankiconnector.cpp:63]: Added an `if` that returns if ankiconnect isn't enabled in settings.

**Pull request merged**